src/main/java/jenkins/branch/NoTriggerBranchProperty.java

```diff
@@ -45,9 +48,45 @@
 @Restricted(NoExternalUse.class)
 public class NoTriggerBranchProperty extends BranchProperty {
 
+    /**
+     * All builds triggered by SCM are suppressed.
+     *
+     * @since 2.6.6
+     */
+    public static final String ALL = "all";
+    /**
+     * Only builds triggered by {@link BranchIndexingCause} are suppressed.
+     *
+     * @since 2.6.6
+     */
+    public static final String INDEXING = "indexing";
+    /**
+     * Only builds triggered by {@link BranchEventCause} are suppressed.
+     *
+     * @since 2.6.6
+     */
+    public static final String EVENTS = "events";
+
+    private String strategyId = ALL;
```

Comment:
Better to use an `enum`.

Reply:
I replaced `static` fields by `enum`. Let me know if it is fine - if yes, I squash the commits.

Reply:
The build has failed in the Windows branch with error:
```
Cannot contact EC2 (aws) - Windows 2019 (i-02c6b5fa1020239a4): hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@67050906:EC2 (aws) - Windows 2019 (i-02c6b5fa1020239a4)": Remote call on EC2 (aws) - Windows 2019 (i-02c6b5fa1020239a4) failed. The channel is closing down or has closed down
```

Reply:
No need to squash BTW. Up to the plugin maintainer where to true-merge or squash-merge a PR. Also best to avoid rebasing with `master` as it interferes with GH reviews; better to true-merge (e.g. `git pull upstream master:master`).

Reply:
I hope somebody squash it on merge because commits like
* add upgrade (migration) tests
* Update src/main/resources/jenkins/branch/Messages.properties

don't bring any value when the git history is analyzed.

Reply:
Marking as resolved. No worries, I prefer squash-merge

Reply:
Marking as resolved. No worries, I prefer squash-merge